revpi_device_info/device_info.py

```diff
@@ -0,0 +1,138 @@
+from datetime import date
+import json
+from typing import List
+
+
+class RevPiHatEEPROMAttributeException(Exception):
+    pass
+
+
+class RevPiDeviceInfo:
+    PRODUCT_ID_BASE = 100000
+
+    def __init__(self, load_contents: bool = True, hat_path: str = "/proc/device-tree/hat/") -> None:
+        """
+        Create new instance of RevPiDeviceInfo.
+
+        :param bool load_contents: Populate class with data from RevPi Hat EEPROM
+        :param str hat_path: Path to HAT files
+        """
+        self._hat_path = hat_path
+
+        self.uuid: str = None
+        self.format_version: int = None
+
+        self.vendor: str = None
+        self.product: str = None
+        self.product_id: int = None
+        self.product_id_revision: str = None
+        self.product_revision: int = None
+        self.product_version: str = None
+        self.product_version_major: int = None
+        self.product_version_minor: int = None
+
+        self.serial: int = None
+        self.eol_date: date = None
+        self.batch_number: int = None
+        self.first_mac_address: str = None
+
+        self._raw_values = {}
+
+        if load_contents:
+            self.load()
+
+    def load(self):
+        """
+        Load values from RevPi HAT EEPROM
+        :raises: RevPiHatEEPROMAttributeException: if the attribute cannot be read from HAT files
+        """
+        self.uuid = self._hat_attribute("uuid")
+        self.format_version = self._hat_attribute_int("custom_0")
+
+        self.vendor = self._hat_attribute("vendor")
+        self.product = self._hat_attribute("product")
+        self.product_id = self._hat_attribute_int("product_id") \
+            + self.PRODUCT_ID_BASE
+        self.product_revision = self._hat_attribute_int("custom_2")
+        self.product_version = self._hat_attribute_version("product_ver")
+        self.product_version_major = self._version_major(self.product_version)
+        self.product_version_minor = self._version_minor(self.product_version)
+        self.product_id_revision = f"PR{self.product_id}R{self.product_revision:02}"
+
+        self.serial = self._hat_attribute_int("custom_1")
+        self.eol_date = self._hat_attribute_date("custom_3")
+        self.batch_number = self._hat_attribute_int("custom_4")
+        self.first_mac_address = self._hat_attribute("custom_5")
+
+    def _version_major(self, version: str) -> int:
+        (major, _) = version.split(".")
+
+        return int(major)
+
+    def _version_minor(self, version: str) -> int:
+        (_, minor) = version.split(".")
+
+        return int(minor)
```

Comment:
Yes, this would be another possibility. If deliberately choose the two method approach as this is more explicit to me ("Explicit is better than implicit.", https://peps.python.org/pep-0020/). Although this means (a little) code duplication